### PR TITLE
[smartplaylist] fix musicvideo grouping by artist

### PR DESCRIPTION
**Pull request by mkortstiege**
This adjusts the playlists group type in case we're retrieving a grouped smartplaylist based on artists. This is needed because the video library is using the actorslink table for artists now. Currently we're returning a completely unfiltered list provided by `GetActorsNav()`.

The second commit is pure cosmetic and removes old/obsolete code.

@Montellese, as discussed i went the re-write route in SmartPlayList.cpp. I think it's less error prone.

Reported in http://forum.kodi.tv/showthread.php?tid=231171&pid=2044997#pid2044997

### CODE REVIEW

**Comment by mkortstiege** [comment]
jenkins build this please

File: xbmc/filesystem/SmartPlaylistDirectory.cpp [hunk 1]
```diff
@@ -233,6 +233,12 @@ namespace XFILE
         if (!videoUrl.FromString(baseDir))
           return false;
 
+        // adjust the group in case we're retrieving a grouped playlist
+        // based on artists. This is needed because the video library
+        // is using the actorslink table for artists.
+        if (isGrouped && group == "artists")
+          mvidPlaylist.SetGroup("actors");
```
Comment (Montellese) [inline xbmc/filesystem/SmartPlaylistDirectory.cpp]: Should the local `group` variable also be set to `"actors"` as it's being used further down?

Comment (mkortstiege) [inline xbmc/filesystem/SmartPlaylistDirectory.cpp]: Not entirely sure what you mean with local `group` variable. Do you mean replacing it in the `.xsp` ?

Comment (Montellese) [inline xbmc/filesystem/SmartPlaylistDirectory.cpp]: In `CSmartPlaylistDirectory::GetDirectory()` (i.e. the method this change is in) there's a local `group` variable which you check against `"artists"` but then you only change the group value in the XSP to `"actors"` and not the cached group value in the local `group` variable which is still `"artists"`.

**Comment by mkortstiege** [comment]
Updated. It's now setting local and smartplaylist group. jenkins build this please.

**Comment by mkortstiege** [comment]
@Montellese i think this is good to go now?

**Comment by Montellese** [comment]
Yeah should be fine.

**Pull request merged**